ddprof-exporter/src/connector/conn_stream.rs

```diff
@@ -0,0 +1,145 @@
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2021-Present Datadog, Inc.
+
+use std::{
+    pin::Pin,
+    task::{Context, Poll},
+};
+
+use futures::{future, Future, FutureExt, TryFutureExt};
+use hyper_rustls::HttpsConnector;
+use pin_project::pin_project;
+
+#[derive(Debug)]
+#[pin_project(project=ConnStreamProj)]
+pub enum ConnStream {
+    Tcp {
+        #[pin]
+        transport: tokio::net::TcpStream,
+    },
+    Tls {
+        #[pin]
+        transport: Box<tokio_rustls::client::TlsStream<tokio::net::TcpStream>>,
```

Comment:
I think that the decision to box or not should be based on how likely we are to be in a situation where Https is not available.
If we’re likely to downgrade to http and use the Tcp variant, the size saving per connection may be meaningful if we open a lot of them (about 500 byte per connection. Are we going to that many, I doubt it though?)
If https is almost always available, we’re just adding an indirection to save space in a minority of cases

Honestly I would be in favor of just ignoring the lint and not boxing, but it probably doesn't matter that much

Reply:
Yeah I've had the same thoughts. 

I opted for boxing instead of ignoring the lint for 2 reasons:
1. Adding an indirection for a IO connection should not affect performance, and ignoring lints IMO is a bad precedent - unless there are clear benefits
2. It actually has some potential for space saving.

tl;dr; potentially single use box, looks to me eyes better than an ignored lint :)

Reply:
I'm ok with Boxing